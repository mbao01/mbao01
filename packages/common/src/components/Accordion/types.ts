import * as AccordionPrimitive from "@radix-ui/react-accordion";

export type AccordionProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Root>;

export type AccordionContentProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Content
>;

export type AccordionItemProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Item>;

export type AccordionTriggerProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Trigger
>;
