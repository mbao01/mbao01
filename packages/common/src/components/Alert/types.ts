import { type VariantProps } from "../../libs";
import { getAlertClasses } from "./constants";

export type AlertProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getAlertClasses>;

export type AlertTitleProps = React.HTMLAttributes<HTMLHeadingElement>;

export type AlertDescriptionProps = React.HTMLAttributes<HTMLDivElement>;
