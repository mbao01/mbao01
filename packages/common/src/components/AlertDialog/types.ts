import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog";
import { type VariantProps } from "../../libs";
import { getButtonClasses } from "../Button/constants";
import {
  getDialogContentClasses,
  getDialogDescriptionClasses,
  getDialogFooterClasses,
  getDialogHeaderClasses,
  getDialogOverlayClasses,
  getDialogTitleClasses,
} from "../Dialog/constants";

export type AlertDialogProps = React.ComponentPropsWithoutRef<typeof AlertDialogPrimitive.Root>;

export type AlertDialogOverlayProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Overlay
> &
  VariantProps<typeof getDialogOverlayClasses>;

export type AlertDialogContentProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Content
> &
  VariantProps<typeof getDialogContentClasses>;

export type AlertDialogHeaderProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogHeaderClasses>;

export type AlertDialogFooterProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogFooterClasses>;

export type AlertDialogTitleProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Title
> &
  VariantProps<typeof getDialogTitleClasses>;

export type AlertDialogDescriptionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Description
> &
  VariantProps<typeof getDialogDescriptionClasses>;

export type AlertDialogActionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;

export type AlertDialogCancelProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;
