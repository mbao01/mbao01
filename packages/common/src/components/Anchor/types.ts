import { VariantProps } from "../../libs";
import { getAnchorClasses } from "./constant";

export type AnchorProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & {
  isExternal?: boolean;
} & VariantProps<typeof getAnchorClasses>;
