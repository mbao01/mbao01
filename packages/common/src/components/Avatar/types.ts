import * as AvatarPrimitive from "@radix-ui/react-avatar";
import { type VariantProps } from "../../libs";
import {
  getAvatarClasses,
  getAvatarFallbackClasses,
  getAvatarImageClasses,
} from "./constants";

export type AvatarProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>,
  "asChild" | "size"
> &
  VariantProps<typeof getAvatarClasses>;

export type AvatarImageProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>,
  "asChild"
> &
  VariantProps<typeof getAvatarImageClasses>;

export type AvatarFallbackProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>,
  "asChild"
> &
  VariantProps<typeof getAvatarFallbackClasses>;
