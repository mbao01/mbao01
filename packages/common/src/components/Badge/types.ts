import { type VariantProps } from "../../helpers";
import { getBadgeClasses } from "./constants";

export type BadgeProps = React.HTMLAttributes<HTMLSpanElement> &
  VariantProps<typeof getBadgeClasses>;
