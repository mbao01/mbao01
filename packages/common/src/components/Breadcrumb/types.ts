import { ReactNode } from "react";

export type BreadcrumbProps = React.HTMLAttributes<HTMLElement> & {
  separator?: ReactNode;
};

export type BreadcrumbListProps = React.HTMLAttributes<HTMLUListElement>;

export type BreadcrumbItemProps = React.HTMLAttributes<HTMLLIElement>;

export type BreadcrumbLinkProps = React.HTMLAttributes<HTMLAnchorElement> & {
  asChild?: boolean;
};

export type BreadcrumbPageProps = React.HTMLAttributes<HTMLSpanElement>;

export type BreadcrumbSeparatorProps = React.HTMLAttributes<HTMLLIElement>;

export type BreadcrumbEllipsisProps = React.HTMLAttributes<HTMLSpanElement>;
