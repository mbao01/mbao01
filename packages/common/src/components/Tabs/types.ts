import * as TabsPrimitive from "@radix-ui/react-tabs";
import { type VariantProps } from "../../libs";
import {
  getTabsContentClasses,
  getTabsListClasses,
  getTabsTriggerClasses,
} from "./constants";

export type TabsProps = React.ComponentPropsWithoutRef<
  typeof TabsPrimitive.Root
>;

export type TabsListProps = React.ComponentPropsWithoutRef<
  typeof TabsPrimitive.List
> &
  VariantProps<typeof getTabsListClasses>;

export type TabsTriggerProps = React.ComponentPropsWithoutRef<
  typeof TabsPrimitive.Trigger
> &
  VariantProps<typeof getTabsTriggerClasses>;

export type TabsContentProps = React.ComponentPropsWithoutRef<
  typeof TabsPrimitive.Content
> &
  VariantProps<typeof getTabsContentClasses>;
