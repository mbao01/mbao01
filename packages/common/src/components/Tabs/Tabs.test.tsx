import { render, screen } from "@testing-library/react";
import { type TabsProps } from "./types";
import { Tabs } from "./";
import userEvent from "@testing-library/user-event";

describe("Tabs", () => {
  const renderTabs = (props?: TabsProps) => {
    return render(
      <Tabs className="w-[400px]" {...props}>
        <Tabs.List className="grid w-full grid-cols-2">
          <Tabs.Trigger value="account">Account</Tabs.Trigger>
          <Tabs.Trigger value="password">Password</Tabs.Trigger>
          <Tabs.Trigger value="secure" disabled>
            Secure
          </Tabs.Trigger>
        </Tabs.List>
        <Tabs.Content value="account">Account tab content</Tabs.Content>
        <Tabs.Content value="password">Password tab content</Tabs.Content>
        <Tabs.Content value="secure">Secure tab content</Tabs.Content>
      </Tabs>
    );
  };

  it("renders tab without a default tab", () => {
    const { asFragment } = renderTabs();
    const triggers = ["Account", "Password", "Secure"];
    const contents = triggers.map((trigger) => `${trigger} tab content`);

    triggers.forEach((trigger) => {
      expect(screen.getByRole("tab", { name: trigger })).toHaveAttribute(
        "data-state",
        "inactive"
      );
    });

    // no tab content shown
    contents.forEach((content) => {
      expect(screen.queryByText(content)).not.toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders tab without a default tab", () => {
    const { asFragment } = renderTabs({ defaultValue: "password" });
    const content = "Password tab content";

    expect(screen.getByRole("tab", { name: "Password" })).toHaveAttribute(
      "data-state",
      "active"
    );

    expect(screen.queryByText(content)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should switch between tabs", async () => {
    const user = userEvent.setup();
    const { asFragment } = renderTabs({ defaultValue: "secure" });

    expect(screen.getByRole("tab", { name: "Secure" })).toBeDisabled();
    expect(screen.getByText("Secure tab content")).toBeInTheDocument();

    await user.click(screen.getByRole("tab", { name: "Password" }));
    expect(screen.getByText("Password tab content")).toBeInTheDocument();

    await user.click(screen.getByRole("tab", { name: "Account" }));
    expect(screen.getByText("Account tab content")).toBeInTheDocument();

    await user.click(screen.getByRole("tab", { name: "Secure" }));
    // "secure" tab is disabled so can't switch to it.
    expect(screen.getByText("Account tab content")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
