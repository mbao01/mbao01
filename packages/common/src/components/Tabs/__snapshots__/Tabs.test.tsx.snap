// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs > renders tab without a default tab 1`] = `
<DocumentFragment>
  <div
    class="w-[400px]"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="h-9 items-center justify-center rounded-lg bg-base-200 p-1 grid w-full grid-cols-2"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-:r0:-content-account"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r0:-trigger-account"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Account
      </button>
      <button
        aria-controls="radix-:r0:-content-password"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r0:-trigger-password"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Password
      </button>
      <button
        aria-controls="radix-:r0:-content-secure"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-disabled=""
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        disabled=""
        id="radix-:r0:-trigger-secure"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Secure
      </button>
    </div>
    <div
      aria-labelledby="radix-:r0:-trigger-account"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r0:-content-account"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-:r0:-trigger-password"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r0:-content-password"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-:r0:-trigger-secure"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r0:-content-secure"
      role="tabpanel"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;

exports[`Tabs > renders tab without a default tab 2`] = `
<DocumentFragment>
  <div
    class="w-[400px]"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="h-9 items-center justify-center rounded-lg bg-base-200 p-1 grid w-full grid-cols-2"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-:r4:-content-account"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r4:-trigger-account"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Account
      </button>
      <button
        aria-controls="radix-:r4:-content-password"
        aria-selected="true"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="active"
        id="radix-:r4:-trigger-password"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Password
      </button>
      <button
        aria-controls="radix-:r4:-content-secure"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-disabled=""
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        disabled=""
        id="radix-:r4:-trigger-secure"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Secure
      </button>
    </div>
    <div
      aria-labelledby="radix-:r4:-trigger-account"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r4:-content-account"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-:r4:-trigger-password"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="active"
      id="radix-:r4:-content-password"
      role="tabpanel"
      style="animation-duration: 0s;"
      tabindex="0"
    >
      Password tab content
    </div>
    <div
      aria-labelledby="radix-:r4:-trigger-secure"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r4:-content-secure"
      role="tabpanel"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;

exports[`Tabs > should switch between tabs 1`] = `
<DocumentFragment>
  <div
    class="w-[400px]"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="h-9 items-center justify-center rounded-lg bg-base-200 p-1 grid w-full grid-cols-2"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-:r8:-content-account"
        aria-selected="true"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="active"
        id="radix-:r8:-trigger-account"
        role="tab"
        tabindex="0"
        type="button"
      >
        Account
      </button>
      <button
        aria-controls="radix-:r8:-content-password"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r8:-trigger-password"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Password
      </button>
      <button
        aria-controls="radix-:r8:-content-secure"
        aria-selected="false"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md px-3 py-1 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-base-100 data-[state=active]:text-foreground data-[state=active]:shadow"
        data-disabled=""
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        disabled=""
        id="radix-:r8:-trigger-secure"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Secure
      </button>
    </div>
    <div
      aria-labelledby="radix-:r8:-trigger-account"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="active"
      id="radix-:r8:-content-account"
      role="tabpanel"
      tabindex="0"
    >
      Account tab content
    </div>
    <div
      aria-labelledby="radix-:r8:-trigger-password"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r8:-content-password"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-:r8:-trigger-secure"
      class="mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
      data-orientation="horizontal"
      data-state="inactive"
      hidden=""
      id="radix-:r8:-content-secure"
      role="tabpanel"
      style=""
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
